fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMutualTest.java

```diff
@@ -976,4 +980,449 @@ void testMultiTargetMultiType() {
+        indexes.add(new Index("indexD", new GroupingKeyExpression(EmptyKeyExpression.EMPTY, 0), IndexTypes.COUNT));
+
+        openSimpleMetaData();
+        int numRecords = 333;
+        populateData(numRecords);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+
+        // Start indexing, crash halfway
+        final List<Tuple> boundariesList = getBoundariesList(numRecords, 4);
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        IntStream.rangeClosed(0, 4).parallel().forEach(id -> {
+            oneThreadIndexingCrashHalfway(indexes, timer, boundariesList, 3);
+        });
+
+        // Block with block-id
+        FDBRecordStoreTestBase.RecordMetaDataHook localHook = allIndexesHook(indexes);
+        openSimpleMetaData(localHook);
+        String luka = "Blocked by Luka";
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setTargetIndexes(indexes)
+                .build()) {
+            indexer.blockIndexBuilds(luka, 10L);
+        }
+
+        // Test query, ensure blocked
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setTargetIndexes(indexes)
+                .build()) {
+            final Map<String, IndexBuildProto.IndexBuildIndexingStamp> stampMap =
+                    indexer.queryIndexingStamps();
+            final List<String> indexNames = indexes.stream().map(Index::getName).collect(Collectors.toList());
+            assertTrue(stampMap.keySet().containsAll(indexNames));
+            for (String indexName : indexNames) {
+                final IndexBuildProto.IndexBuildIndexingStamp stamp = stampMap.get(indexName);
+                assertTrue(stamp.getTargetIndexList().containsAll(indexNames));
+                assertEquals(IndexBuildProto.IndexBuildIndexingStamp.Method.MUTUAL_BY_RECORDS, stamp.getMethod());
+                assertTrue(stamp.getBlock());
+                assertEquals(luka, stamp.getBlockID());
+                assertTrue(stamp.getBlockExpireEpochMilliSeconds() > System.currentTimeMillis());
+                assertTrue(stamp.getBlockExpireEpochMilliSeconds() < 20_000 + System.currentTimeMillis());
+            }
+        }
+
+        // Attempt continue while blocked with the wrong id, ensure failure
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setTargetIndexes(indexes)
+                .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                        .setMutualIndexingBoundaries(boundariesList)
+                        .setAllowUnblock(true, "Blocked by Leonardo")
```

Comment:
This tests trying to start an index build with the wrong block id,  but it would probably be good to call `unblockIndexBuilds` with the wrong id and make sure that doesn't work, and gives back right block info.